Hudson private security realm implementation

divyasivasamy opened the pull request:
HudsonPrivateSecurityRealm (Jenkins Core class) uses a bcrypt algorithm to hash passwords for security reasons which is not a FIPS compliant one. Implemented PBKDF2withHmacSHA512 version of password hashing as an installation/configuration mode.

Testing done

Proposed changelog entries
N/A

Proposed upgrade guidelines
N/A

dependabot[bot] committed:
Bump org.jenkins-ci:jenkins from 1.103 to 1.104 (#8346)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

janfaracik committed:
[JENKINS-71238] New login page breaks `login-theme-plugin` (#8341)

Move simple-header out

jenkins-release-bot committed:
[maven-release-plugin] prepare release jenkins-2.418

jenkins-release-bot committed:
[maven-release-plugin] prepare for next development iteration

NotMyFault committed:
Update release chat room URL (#8348)

renovate[bot] committed:
Update dependency eslint-config-prettier to v9 (#8350)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency postcss-preset-env to v9.1.1 (#8347)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

divyasivasamy committed:
isHashValid Changes

dependabot[bot] committed:
Bump org.jenkins-ci.main:jenkins-test-harness from 2044.v03c87927ff5c to 2053.v0ea_6fc5d99b_f (#8349)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

renovate[bot] committed:
Update jenkins/ath Docker tag to v5688 (#8352)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump bridge-method-injector.version from 1.27 to 1.28 (#8353)

Bumps `bridge-method-injector.version` from 1.27 to 1.28.

Updates `com.infradna.tool:bridge-method-injector` from 1.27 to 1.28
- [Release notes](https://github.com/jenkinsci/bridge-method-injector/releases)
- [Commits](https://github.com/jenkinsci/bridge-method-injector/compare/bridge-method-injector-parent-1.27...bridge-method-injector-parent-1.28)

Updates `com.infradna.tool:bridge-method-annotation` from 1.27 to 1.28
- [Release notes](https://github.com/jenkinsci/bridge-method-injector/releases)
- [Commits](https://github.com/jenkinsci/bridge-method-injector/compare/bridge-method-injector-parent-1.27...bridge-method-injector-parent-1.28)

---
updated-dependencies:
- dependency-name: com.infradna.tool:bridge-method-injector
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: com.infradna.tool:bridge-method-annotation
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

renovate[bot] committed:
Update babel monorepo to v7.22.10 (#8360)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update jenkins/ath Docker tag to v5689 (#8358)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

dten committed:
Use standard size node icon even with long node names (#8089)

SujathaH committed:
Replace MD5 digest with SHA256 while logging payload in Mock class called from tests (#8319)

NotMyFault committed:
Replace Java 19 with 21 in Jenkinsfile (#8361)

dependabot[bot] committed:
Bump org.jenkins-ci.plugins:cloudbees-folder from 6.815.v0dd5a_cb_40e0e to 6.846.v23698686f0f6 (#8359)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

renovate[bot] committed:
Update jenkins/ath Docker tag to v5691 (#8364)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

basil committed:
Add Java 21 to supported Java versions (#8365)

renovate[bot] committed:
Update dependency sass to v1.65.1 (#8366)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

divyasivasamy committed:
isHashValid Changes

renovate[bot] committed:
Update dependency eslint to v8.47.0 (#8373)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.jenkins-ci.plugins:matrix-project from 802.v8013b_40c7edc to 808.v5a_b_5f56d6966 (#8374)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

renovate[bot] committed:
Update jenkins/ath Docker tag to v5695 (#8369)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

jenkins-release-bot committed:
[maven-release-plugin] prepare release jenkins-2.419

jenkins-release-bot committed:
[maven-release-plugin] prepare for next development iteration

janfaracik committed:
Deprecate `findAncestor` and `findAncestorClass` (#8357)

janfaracik committed:
Remove the addition of the 'has-help' class from `hudson-behaviour.js` (#8355)

janfaracik committed:
Update the design of the content blocks (#8363)

Co-authored-by: Tim Jacomb <timjacomb1@gmail.com>
Co-authored-by: Tim Jacomb <21194782+timja@users.noreply.github.com>

janfaracik committed:
Display a notice when there are plugins installed or updates available (#8208)

Co-authored-by: Alexander Brandes <brandes.alexander@web.de>
Co-authored-by: Alexander Brandes <mc.cache@web.de>
Co-authored-by: Tim Jacomb <21194782+timja@users.noreply.github.com>

renovate[bot] committed:
Update jenkins/ath Docker tag to v5699 (#8377)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.jenkins-ci.plugins:cloudbees-folder from 6.846.v23698686f0f6 to 6.848.ve3b_fd7839a_81 (#8383)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.jenkins-ci.plugins:script-security from 1264.vecf66020eb_7d to 1269.v639888f5e366 (#8384)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

divyasivasamy committed:
commented logger

divyasivasamy committed:
commented logger

Santhosh-CloudB commented on core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java:
```diff
@@ -925,49 +934,173 @@ public boolean isHashValid(String hash) {
+
+        @SuppressFBWarnings(value = "MS_SHOULD_BE_FINAL", justification = "Accessible via System Groovy Scripts")
+        @Restricted(NoExternalUse.class)
+        private static int MAXIMUM_PBKDF2_LOG_ROUND = SystemProperties.getInteger(HudsonPrivateSecurityRealm.class.getName() + ".maximumPBKDF2LogRound", (int) Math.pow(2, 10));
```
Still reference for iteration count is pow(2...).. this can be explicit count

Santhosh-CloudB commented on core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java:
```diff
@@ -925,49 +934,173 @@ public boolean isHashValid(String hash) {
+                    return true;
+                }
+            }
+            return false;
```
Exceptions can be caught and logged

Santhosh-CloudB commented on test/src/test/java/hudson/security/HudsonPrivateSecurityRealmTest.java:
```diff
@@ -86,13 +97,6 @@ public void linkExtension() {
     }
 
-    @Before
-    public void setup() throws Exception {
```
Setup has to be global. pls remove local variables and bring it back to global

Santhosh-CloudB commented on test/src/test/java/hudson/security/HudsonPrivateSecurityRealmTest.java:
```diff
@@ -498,11 +512,76 @@ public void controlCharacterAreNoMoreValid_CustomRegex() throws Exception {
+        mockStatic(HudsonPrivateSecurityRealm.class);
+        when(getPasswordHeader()).thenReturn("$PBKDF2");
+        assertNotNull(pbkdf2PasswordEncoder.encode("password"));
+        assertTrue(PASSWORD_ENCODER.isPasswordHashed("$PBKDF2" + PASSWORD_HASH_ENCODER.encode("password")));
```
Negative scenario also needs to be tested

divyasivasamy committed:
modified test cases

janfaracik committed:
Update link design (#8375)

Co-authored-by: Alexander Brandes <mc.cache@web.de>

dependabot[bot] committed:
Bump org.jenkins-ci.plugins:structs from 324.va_f5d6774f3a_d to 325.vcb_307d2a_2782 (#8389)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

divyasivasamy committed:
modified test cases

divyasivasamy committed:
modified test cases

mustafau committed:
Turkish localization fixes for the job configuration page (#8368)

yaroslavafenkin committed:
[JENKINS-71514] Remove `eval` call in hudson-behavior.js (#8370)

Co-authored-by: Yaroslav Afenkin <yaroslavafenkin@users.noreply.github.com>

Mysterious-Dev committed:
Many French translations for plugin panels (#8382)

Co-authored-by: Olivier Lamy <olamy@apache.org>

daniel-beck committed:
[JENKINS-71833] Fix link to job in slow trigger admin monitor (#8387)

Co-authored-by: Daniel Beck <daniel-beck@users.noreply.github.com>

janfaracik committed:
Remove unused CSS variables (#8354)

renovate[bot] committed:
Update dependency prettier to v3.0.2 (#8396)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency postcss to v8.4.28 (#8398)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.jenkins-ci.plugins:display-url-api from 2.3.8 to 2.3.9 (#8397)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

janfaracik committed:
Remove 'track-mouse' JS (#8391)

janfaracik committed:
Remove unused CSS classes (#8372)

Co-authored-by: Daniel Beck <1831569+daniel-beck@users.noreply.github.com>
Co-authored-by: Tim Jacomb <21194782+timja@users.noreply.github.com>

jglick committed:
[JENKINS-61452] Tolerate corrupt Base64 in `PlainTextConsoleOutputStream` (#8378)

renovate[bot] committed:
Update Yarn to v3.6.2 (#8401)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Alexander Brandes <mc.cache@web.de>

renovate[bot] committed:
Update dependency postcss-scss to v4.0.7 (#8400)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.jenkins-ci.plugins:script-security from 1269.v639888f5e366 to 1271.vdede89739a_81 (#8408)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.jenkins-ci.main:jenkins-test-harness from 2053.v0ea_6fc5d99b_f to 2058.va_7b_41a_286207 (#8409)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency sass to v1.66.0 (#8402)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.apache.ant:ant from 1.10.13 to 1.10.14 (#8405)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.springframework.security:spring-security-bom from 5.8.5 to 5.8.6 (#8407)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

janfaracik committed:
Move app bar for plugins (#8376)

* Move app bar for plugins, make sidebar sticky

* Update _updateSite.js

* Remove sticky side panel

daniel-beck committed:
[JENKINS-71848] Remove admin monitors popup from `/manage/` again (#8388)

[JENKINS-71848] Remove admin monitors popup from /manage/ again

Co-authored-by: Daniel Beck <daniel-beck@users.noreply.github.com>

mawinter69 committed:
Replace hetero-list YUI button and menu with new style button and tippy.js menu (#8340)

* refresh hetero-list UI

use new button style
replace the YUI menu with a tippy menu

* fix lint errors

* avoid blanks

* prettier

* keep button in div

* fix test

* fix test #2

* convert inputs to buttons for plugins

* prettier

* remove unnecessary style definition

* remove adjunct

* adjust filter

---------

Co-authored-by: Tim Jacomb <21194782+timja@users.noreply.github.com>

basil committed:
Revert "Replace hetero-list YUI button and menu with new style button and tippy.js menu" (#8412)

Revert "Replace hetero-list YUI button and menu with new style button and tippy.js menu (#8340)"

This broke:

- org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SecureGroovyScriptTest
- hudson.plugins.textfinder.TextFinderPublisherFreestyleTest#createTextFinderViaWebClient

This reverts commit c1a6c6c9ddf432e8eae3db68d3ee13c1f762d785.

renovate[bot] committed:
Update dependency sass to v1.66.1 (#8411)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

divyasivasamy committed:
Merge branch 'master' into HudsonPrivateSecurityRealmImplementation

divyasivasamy committed:
resolved Util conflicts

divyasivasamy committed:
resolved Util conflicts

divyasivasamy committed:
resolved Util conflicts

divyasivasamy committed:
modified test cases

jenkins-release-bot committed:
[maven-release-plugin] prepare release jenkins-2.420

jenkins-release-bot committed:
[maven-release-plugin] prepare for next development iteration

dependabot[bot] committed:
Bump org.jenkins-ci.plugins:matrix-auth from 3.1.10 to 3.2 (#8415)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Vlatombe committed:
Optimize Label parsing calls (#8395)

* Optimize Label#parse for simple label

For a simple label, this speeds up lookup of an existing simple label (no expression used):  16ns vs 106ns (6x faster)

* Fix reviews

* Fix include

* Avoid parsing label every time getAssignedLabels is called

* Store Set<LabelAtom> at Node descendants level
* Compute it whenever labelString is updated

* Fix spotbugs / test issue

* Make it restricted

janfaracik committed:
Refactor SCSS structure (#8410)

Init

daniel-beck committed:
[JENKINS-71087] Nicer 404 error pages (#8250)

* Nicer 404 error pages

* Fix test, expose custom error message on 404 pages

* Record the current user to show an appropriate 404 page

* 404 simple page for the resource domain

the resource domain has no access to the static resources like
javascript and css. Open the images in the filter so we can at least
show an image.
Do not show the hint about not found or permission when accessing the
static-files url from the regular domain.

* Minor fixes and improvements

- Be more restrictive, only allow the specific image on simple 404
- Make extra resources work with non-empty resources path
- Do not redirect to 404 page after login from regular 404 page
- Remove 'Error:' prefix for friendlier page appearance
- Include favicon etc. resources in simple 404 page

* Fall back to 'Not Found' when accessing error pages directly

Also reformat

* Add 404 page tests

* Revert change to ResourceDomainTest

* Move test out of JenkinsTest, make parameterized for different contexts

* Return to the 404'd page after login by using 'from' session attribute

* Prevent 'REST API' link from showing on the error page

* Assert specific message on resource domain

* Remove obsolete commented out line

* Add Javadoc

---------

Co-authored-by: Daniel Beck <daniel-beck@users.noreply.github.com>
Co-authored-by: Markus Winter <m.winter@sap.com>

basil committed:
Simplify `fireEvent` (#8406)

basil committed:
`hudson.model.CauseTest#broadlyNestedCauses` flaked (#8417)

dwnusbaum committed:
[JENKINS-71479] Do not use SCSS lighten function directly to avoid invalid CSS (#8425)

dependabot[bot] committed:
Bump org.jenkins-ci.plugins:script-security from 1271.vdede89739a_81 to 1273.v66c1964f0dfd (#8424)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

basil committed:
`CauseTest#deeplyNestedCauses` is also flaky (#8430)

basil committed:
`MaintainCanTakeStrengtheningTest` is flaky (#8429)

daniel-beck committed:
Fix PCT problem of #8250 (#8426)

Co-authored-by: Daniel Beck <daniel-beck@users.noreply.github.com>

renovate[bot] committed:
Update dependency node to v18.17.1 (#8433)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

dependabot[bot] committed:
[JENKINS-38520] Every message in agent log appears twice with the `-agentLog` option (#8416)

Bump org.jenkins-ci.main:remoting

Bumps [org.jenkins-ci.main:remoting](https://github.com/jenkinsci/remoting) from 3142.vcfca_0cd92128 to 3148.v532a_7e715ee3.
- [Release notes](https://github.com/jenkinsci/remoting/releases)
- [Changelog](https://github.com/jenkinsci/remoting/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jenkinsci/remoting/commits)

---
updated-dependencies:
- dependency-name: org.jenkins-ci.main:remoting
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

daniel-beck committed:
Minor correction in UpdateCenter restart UI internals (#8419)

Co-authored-by: Daniel Beck <daniel-beck@users.noreply.github.com>

daniel-beck committed:
Rename adjunct files to more easily associate with elements (#8423)

Co-authored-by: Daniel Beck <daniel-beck@users.noreply.github.com>

renovate[bot] committed:
Update dependency @babel/core to v7.22.11 (#8439)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.jenkins-ci.main:jenkins-test-harness from 2058.va_7b_41a_286207 to 2062.v3efc79721e45 (#8432)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

renovate[bot] committed:
Update Yarn to v3.6.3 (#8437)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Alexander Brandes <mc.cache@web.de>

dependabot[bot] committed:
Bump com.puppycrawl.tools:checkstyle from 10.12.2 to 10.12.3 (#8441)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

divyasivasamy committed:
fixed review comments

divyasivasamy committed:
fixed review comments

renovate[bot] committed:
Update dependency eslint to v8.48.0 (#8443)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

timja committed:
Add `Appearance` system configuration page (#8403)

* Add `Appearance` system configuration page

* Use one-column layout

* Add developer guidance

basil committed:
Clear the queue before tearing down tests (#8428)

Co-authored-by: Daniel Beck <1831569+daniel-beck@users.noreply.github.com>

divyasivasamy committed:
fixed review comments

jenkins-release-bot committed:
[maven-release-plugin] prepare release jenkins-2.421

jenkins-release-bot committed:
[maven-release-plugin] prepare for next development iteration

Vlatombe committed:
Initialize default view earlier in the init process. (#8413)

* Initialize default view earlier in the init process.

Perform the default view initialization next to loading the
configuration from disk.

This is more consistent as the primary view and views are cleared just
before loading from disk, which leads to a temporary inconsistent state while
Jenkins initializes.

* Update core/src/main/java/jenkins/model/Jenkins.java

Co-authored-by: Jesse Glick <jglick@cloudbees.com>

* View initialization should happen even when the config file doesn't exist.

* Add a test

---------

Co-authored-by: Jesse Glick <jglick@cloudbees.com>

StefanSpieker committed:
Fixed some deprecation warnings in tests (#8434)

fixed some deprecation warnings

StefanSpieker committed:
Fixed some minor deprecation warnings in Tests (#8436)

fixed some deprecation warnings

MarkEWaite committed:
Add more operating system end of life dates (#8438)

Users have reported bugs on additional operating systems that are already
end of life, like Ubuntu 20.10.  Include outdated Ubuntu operating
systems in the list so that the users are warned.

https://endoflife.date/ubuntu

Red Hat Enterprise Linux 9 has released along with its derivatives,
AlmaLinux, Rocky Linux, and Oracle Linux.  Include them in the end of
life data.

https://endoflife.date/rhel

https://endoflife.date/almalinux
https://endoflife.date/oraclelinux
https://endoflife.date/rocky-linux

basil committed:
Side-port test cleanup from jenkinsci/remoting#587 (#8414)

dependabot[bot] committed:
Bump org.jenkins-ci.main:jenkins-test-harness from 2062.v3efc79721e45 to 2064.vcd3b_b_8f3f2b_a_ (#8444)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.jenkins-ci:jenkins from 1.104 to 1.105 (#8445)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mawinter69 committed:
Replace `hetero-list` YUI button and menu with new style button and Tippy.js menu (#8418)

Co-authored-by: Tim Jacomb <21194782+timja@users.noreply.github.com>

Vlatombe committed:
In case an extension of Slave overrides readResolve but does not call its super implementation

recover from it, and issue a warning to raise a bug against the impacted plugin.

Vlatombe committed:
Add a test

Vlatombe committed:
Fix spotbugs/checkstyle

Vlatombe committed:
Apply suggestions from code review

Co-authored-by: Daniel Beck <1831569+daniel-beck@users.noreply.github.com>

daniel-beck committed:
Add telemetry for security-related settings (#8440)

* Add telemetry for security-related settings

* Actually include component information

* FindBugs, also TcpSlaveAgentListener can be null

---------

Co-authored-by: Daniel Beck <daniel-beck@users.noreply.github.com>

Vlatombe committed:
Move test to JenkinsSessionRule

Vlatombe committed:
Rename class

renovate[bot] committed:
Update dependency jquery to v3.7.1 (#8450)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency webpack-remove-empty-scripts to v1.0.4 (#8452)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.jenkins-ci.plugins:script-security from 1273.v66c1964f0dfd to 1274.v2b_33362a_f2f5 (#8451)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency prettier to v3.0.3 (#8454)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.jenkins-ci.plugins:script-security from 1273.v66c1964f0dfd to 1275.v23895f409fb_d (#8455)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

basil committed:
[JENKINS-71950] List plugins in deterministic order (#8453)

List plugins in deterministic order

olamy committed:
Upgrade to winstone 6.13 which upgrades to Jetty 10.0.16 (#8447)

Signed-off-by: Olivier Lamy <olamy@apache.org>

mustafau committed:
Turkish localization fixes for the new job page (#8446)

Vlatombe committed:
Merge pull request #8448 from Vlatombe/labelAtomSet-robustness

Label parsing robustness fix

renovate[bot] committed:
Update dependency @babel/preset-env to v7.22.14 (#8458)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency postcss to v8.4.29 (#8456)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency postcss-preset-env to v9.1.2 (#8449)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

muminkoykiran committed:
[JENKINS-71958] Update scriptConsole_tr.properties (#8460)

Update scriptConsole_tr.properties

* The Turkish translation of the "Run" parameter has been corrected.

dependabot[bot] committed:
Bump actions/checkout from 3 to 4 (#8463)

Bumps [actions/checkout](https://github.com/actions/checkout) from 3 to 4.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

jenkins-release-bot committed:
[maven-release-plugin] prepare release jenkins-2.422

jenkins-release-bot committed:
[maven-release-plugin] prepare for next development iteration

divyasivasamy committed:
fixed review comments

divyasivasamy committed:
fixed review comments

daniel-beck committed:
[SECURITY-3261]

yaroslavafenkin committed:
[SECURITY-3245]

Kevin-CB committed:
[SECURITY-3072]

Kevin-CB committed:
[SECURITY-3073]

dependabot[bot] committed:
Bump io.jenkins.plugins:commons-text-api from 1.10.0-68.v0d0b_c439292b_ to 1.10.0-78.v3e7b_ea_d5a_fe1 (#8464)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency postcss-preset-env to v9.1.3 (#8466)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump antlr.version from 4.13.0 to 4.13.1 (#8465)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

jtnord committed:
Apply suggestions from code review

jtnord committed:
checkstyle

jtnord committed:
introduce specific class for FIPS

jtnord committed:
Merge remote-tracking branch 'origin/master' into HudsonPrivateSecurityRealmImplementation

jtnord committed:
misc code cleanups

jtnord committed:
revert changes to HudsonPrivateSecurityRealmTest

renovate[bot] committed:
Update babel monorepo to v7.22.15 (#8470)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

jtnord committed:
code fixups and moving tests

jtnord committed:
code fixups and moving tests

jtnord committed:
make methods and fields private, and add message to exception

jtnord committed:
SecureRandom s -> SecureRandoms

jtnord committed:
fixup IT after increasing the number of hasing rounds

basil committed:
Avoid usage of Guava from `ProcessTree` (#8467)

* Avoid usage of Guava from `ProcessTree`

* Update core/src/main/java/hudson/util/ProcessTree.java

Co-authored-by: Vincent Latombe <vincent@latombe.net>

---------

Co-authored-by: Vincent Latombe <vincent@latombe.net>

mawinter69 committed:
update badge documentation (#8469)

* badge documentation

Include in the badge documentation that it can be used in <l:task links

Set type for the badge attribute of lib/form/task

* fix javadoc

Vlatombe committed:
Make branding overridable in signin/login page (#8462)

* Move resources from DefaultSimplePageDecorator to SimplePageDecorator to ease implementation.

janfaracik committed:
Move Node monitoring option to app bar (#8381)

* Init

* Update index.jelly

* Update index.jelly

* Add description

* Add description

* Fix test

* Update ComputerSetTest.java

---------

Co-authored-by: Alexander Brandes <mc.cache@web.de>

Kevin-CB committed:
Add test for SECURITY-3135 (#8427)

* Add test for SECURITY-3135

* Address review feedback

mawinter69 committed:
Prevent scrolling for textarea preview (#8472)

prevent scrolling for preview

Prevent that when clicking on preview for a textarea, the page scrolls
to the top and the url changed.

mawinter69 committed:
[JENKINS-71983] allow symbols in MenuItem#withIconClass (#8471)

[JENKINS-71983] fix symbols in MenuItem#withIconClass

when a symbol is passed to MenuItem#withIconClass the symbol is now
properly displayed

jtnord committed:
move SecureRandom to double-checked-locking as SecureRandom itself is thread safe

renovate[bot] committed:
Update dependency postcss-scss to v4.0.8 (#8475)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump slf4jVersion from 2.0.7 to 2.0.9 (#8461)

Bumps `slf4jVersion` from 2.0.7 to 2.0.9.

Updates `org.slf4j:jcl-over-slf4j` from 2.0.7 to 2.0.9

Updates `org.slf4j:log4j-over-slf4j` from 2.0.7 to 2.0.9

Updates `org.slf4j:slf4j-api` from 2.0.7 to 2.0.9

Updates `org.slf4j:slf4j-jdk14` from 2.0.7 to 2.0.9

---
updated-dependencies:
- dependency-name: org.slf4j:jcl-over-slf4j
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: org.slf4j:log4j-over-slf4j
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: org.slf4j:slf4j-api
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: org.slf4j:slf4j-jdk14
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump com.github.eirslett:frontend-maven-plugin from 1.13.4 to 1.14.0 (#8479)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.apache.commons:commons-compress from 1.23.0 to 1.24.0 (#8478)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

jtnord committed:
make FIPS_COMPLIANCE_MODE  final

changing this at run time makes no sense as decisions will already have been made to configure things differently.

jenkins-release-bot committed:
[maven-release-plugin] prepare release jenkins-2.423

jenkins-release-bot committed:
[maven-release-plugin] prepare for next development iteration

yaroslavafenkin committed:
Merge branch 'security-master'

renovate[bot] committed:
Update dependency eslint to v8.49.0 (#8480)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency @babel/core to v7.22.17 (#8476)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

jenkins-release-bot committed:
[maven-release-plugin] prepare release jenkins-2.424

jenkins-release-bot committed:
[maven-release-plugin] prepare for next development iteration

yaroslavafenkin committed:
Merge branch 'master' into security-master

jglick committed:
`StringIndexOutOfBoundsException` in `AbstractItem.getUrl` (#8481)

basil committed:
Increase minimum required Remoting version from 4.7 to 4.13 (#8484)

basil committed:
Migrate from deprecated finalization API to Java 9+ API (#8486)

* Migrate from deprecated finalization API to Java 9+ API

* Use non-deprecated JTH method

basil committed:
Remove usages of deprecated `ThreadDeath` (#8487)

basil committed:
Remove finalization from `RegistryKey` (#8488)

Vlatombe committed:
LogRecorder programmatic deletion support (#8489)

* Add programmatic deletion support for LogRecorder.
* Also added event handling through SaveableListener whenever it gets deleted.

jglick committed:
[JENKINS-72016] Define a thread pool for `ProxyConfiguration`’s `HttpClient` (#8490)

* Define a thread pool for `ProxyConfiguration`’s `HttpClient`

* Skip test on Windows https://github.com/jenkinsci/jenkins/pull/8490#issuecomment-1720108881

mawinter69 committed:
[JENKINS-71880] fix notification in breadcrumb (#8493)

* [JENKINS-71880] fix notification in breadcrumb

fix the notification when e.g. triggering a build via the breadcrumb.

replace the hoverNotification with a notificationBar. When the
hoverNotification appears the actual menu disappears making the
notification somehow look misplaced.
Prepend the message with the actual menu entry to make it more clear
what the message means.

* run prettier

jglick committed:
Optimization for `AbstractLazyLoadRunMap.loadNumberOnDisk` (#8494)

renovate[bot] committed:
Update dependency sass to v1.67.0 (#8495)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

abhishekmaity committed:
Add the 'alt' attribute to the logo in the README.md (#8497)

renovate[bot] committed:
Update babel monorepo to v7.22.20 (#8498)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.jenkins-ci.plugins:matrix-auth from 3.2 to 3.2.1 (#8499)

Bumps [org.jenkins-ci.plugins:matrix-auth](https://github.com/jenkinsci/matrix-auth-plugin) from 3.2 to 3.2.1.
- [Release notes](https://github.com/jenkinsci/matrix-auth-plugin/releases)
- [Changelog](https://github.com/jenkinsci/matrix-auth-plugin/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jenkinsci/matrix-auth-plugin/compare/matrix-auth-3.2...matrix-auth-3.2.1)

---
updated-dependencies:
- dependency-name: org.jenkins-ci.plugins:matrix-auth
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump io.jenkins.plugins:font-awesome-api from 6.4.0-2 to 6.4.2-1 (#8500)

Bumps [io.jenkins.plugins:font-awesome-api](https://github.com/jenkinsci/font-awesome-api-plugin) from 6.4.0-2 to 6.4.2-1.
- [Release notes](https://github.com/jenkinsci/font-awesome-api-plugin/releases)
- [Changelog](https://github.com/jenkinsci/font-awesome-api-plugin/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jenkinsci/font-awesome-api-plugin/compare/v6.4.0-2...v6.4.2-1)

---
updated-dependencies:
- dependency-name: io.jenkins.plugins:font-awesome-api
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

PierreBtz committed:
[chore] Remove useless TODO (#8501)

basil committed:
Use functionality from `XStream2` where possible (#8504)

eltociear committed:
Fix typo in container-debug.js (#8506)

documenation -> documentation

dependabot[bot] committed:
Bump org.springframework.security:spring-security-bom from 5.8.6 to 5.8.7 (#8502)

Bump org.springframework.security:spring-security-bom

Bumps [org.springframework.security:spring-security-bom](https://github.com/spring-projects/spring-security) from 5.8.6 to 5.8.7.
- [Release notes](https://github.com/spring-projects/spring-security/releases)
- [Changelog](https://github.com/spring-projects/spring-security/blob/main/RELEASE.adoc)
- [Commits](https://github.com/spring-projects/spring-security/compare/5.8.6...5.8.7)

---
updated-dependencies:
- dependency-name: org.springframework.security:spring-security-bom
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

jtnord committed:
fix signup autofocus (#8474)

the expression was not valid and caused an exception to be thrown.

replace the invalid expression with a null check (for the initial
rendering where there is no SignupIno in the page).

```
  10.912 [id=100]       WARNING h.ExpressionFactory2$JexlExpression#evaluate: Caught exception evaluating: (data.errors.containsKey('username') || !data) ? true : null in /jenkins/securityRealm/createAccount. Reason: java.lang.Exception: expression not boolean valued
java.lang.Exception: expression not boolean valued
        at org.apache.commons.jexl.parser.ASTNotNode.value(ASTNotNode.java:68)
...
        at org.kohsuke.stapler.ScriptRequestDispatcher.forward(ScriptRequestDispatcher.java:97)
        at hudson.security.HudsonPrivateSecurityRealm.createAccount(HudsonPrivateSecurityRealm.java:399)
        at hudson.security.HudsonPrivateSecurityRealm._doCreateAccount(HudsonPrivateSecurityRealm.java:269)
        at hudson.security.HudsonPrivateSecurityRealm.doCreateAccount(HudsonPrivateSecurityRealm.java:261)
```

renovate[bot] committed:
Update jenkins/ath Docker tag to v5717 (#8518)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency postcss-preset-env to v9.1.4 (#8517)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency postcss to v8.4.30 (#8516)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

jglick committed:
Ignore `ProxyConfigurationTest.httpClientExecutor` (#8512)

Kevin-CB committed:
Explicit log levels (#8509)

* improve-logs-level-clarity

* more verbose instead of lower

dependabot[bot] committed:
Bump org.jenkins-ci.plugins:junit from 1217.v4297208a_a_b_ce to 1240.vf9529b_881428 (#8507)

Bump org.jenkins-ci.plugins:junit

Bumps [org.jenkins-ci.plugins:junit](https://github.com/jenkinsci/junit-plugin) from 1217.v4297208a_a_b_ce to 1240.vf9529b_881428.
- [Release notes](https://github.com/jenkinsci/junit-plugin/releases)
- [Commits](https://github.com/jenkinsci/junit-plugin/commits)

---
updated-dependencies:
- dependency-name: org.jenkins-ci.plugins:junit
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

RodrigoRVSN committed:
fix(typo): adjust typos in pt br translation of plugin setup wizard (#8514)

mustafau committed:
Turkish localizations for the clouds page (#8510)

mawinter69 committed:
[JENKINS-71866] Allow to use alternate values for "Build with Parameters" and the "Build" button on the parameters page (#8496)

[JENKINS-71866] alternate for Build with Parameters

Allow to use alternate values for "Build with Parameters" and the
"Build" button on the parameters page.
See also JENKINS-46554 and JENKINS-27214

jtnord committed:
Add (c) header

jtnord committed:
keep the checkstyle overlord happy

jtnord committed:
update exception message depending on the specific error issue

renovate[bot] committed:
Update dependency sass to v1.68.0 (#8521)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

basil committed:
Fix compilation error with `maven.compiler.release=21` (#8520)

mustafau committed:
Turkish localization for "Build with Parameters" link (#8519)

flabrie committed:
[JENKINS-72018] Form entries in .repeated-chunk using .show-if-not-only CSS class are not hidden when they are alone (#8491)

* -	Add the missing definition for the .only CSS class inside of .repeated-chunk .show-if-not-only ones. Without these CSS classes, form entries using .show-if-not-only class are not hidden when alone.

* Update war/src/main/scss/form/_reorderable-list.scss

Co-authored-by: Daniel Beck <1831569+daniel-beck@users.noreply.github.com>

* Update war/src/main/scss/form/_reorderable-list.scss

Co-authored-by: Daniel Beck <1831569+daniel-beck@users.noreply.github.com>

* Update war/src/main/scss/form/_reorderable-list.scss

Co-authored-by: Daniel Beck <1831569+daniel-beck@users.noreply.github.com>

---------

Co-authored-by: Francis Labrie <francis.labrie@cgi.com>
Co-authored-by: Daniel Beck <1831569+daniel-beck@users.noreply.github.com>

benebsiny committed:
[JENKINS-71890] Model links in build history views and administrative monitors no longer work (#8485)

dependabot[bot] committed:
Bump stapler.version from 1802.1804.va_8d30483a_7f7 to 1814.vdc9dd5217ee2 (#8522)

Bump stapler.version

Bumps `stapler.version` from 1802.1804.va_8d30483a_7f7 to 1814.vdc9dd5217ee2.

Updates `org.kohsuke.stapler:stapler` from 1802.1804.va_8d30483a_7f7 to 1814.vdc9dd5217ee2
- [Release notes](https://github.com/jenkinsci/stapler/releases)
- [Changelog](https://github.com/jenkinsci/stapler/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jenkinsci/stapler/commits)

Updates `org.kohsuke.stapler:stapler-groovy` from 1802.1804.va_8d30483a_7f7 to 1814.vdc9dd5217ee2
- [Release notes](https://github.com/jenkinsci/stapler/releases)
- [Changelog](https://github.com/jenkinsci/stapler/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jenkinsci/stapler/commits)

---
updated-dependencies:
- dependency-name: org.kohsuke.stapler:stapler
  dependency-type: direct:production
- dependency-name: org.kohsuke.stapler:stapler-groovy
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Vlatombe committed:
Logging XmlFile write to help track race conditions during tests. (#8513)

jtnord committed:
Leave comment about commented out tests

jtnord committed:
remove extra space in comment

jtnord committed:
remove trailing line ends to keep spotless happy

jenkins-release-bot committed:
[maven-release-plugin] prepare release jenkins-2.425

jenkins-release-bot committed:
[maven-release-plugin] prepare for next development iteration

dependabot[bot] committed:
Bump org.apache.maven.plugins:maven-shade-plugin from 3.5.0 to 3.5.1 (#8524)

Bumps [org.apache.maven.plugins:maven-shade-plugin](https://github.com/apache/maven-shade-plugin) from 3.5.0 to 3.5.1.
- [Release notes](https://github.com/apache/maven-shade-plugin/releases)
- [Commits](https://github.com/apache/maven-shade-plugin/compare/maven-shade-plugin-3.5.0...maven-shade-plugin-3.5.1)

---
updated-dependencies:
- dependency-name: org.apache.maven.plugins:maven-shade-plugin
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Vlatombe committed:
Add data-bound setters for View (#8511)

* Add data-bound setters for View

This allows programmatic edit of View

* Clean up

* Fix test

---------

Co-authored-by: Alexander Brandes <mc.cache@web.de>

StefanSpieker committed:
Use `isEmpty()` and simplified assertions in Test (#8473)

* use isEmpty() and simplified assertions

* use `assertThat` instead of `assertFalse` to get better messages in case of failure

* Update test/src/test/java/lib/hudson/ListScmBrowsersTest.java

Co-authored-by: James Nord <jtnord@users.noreply.github.com>

* Update test/src/test/java/hudson/PluginManagerTest.java

Co-authored-by: James Nord <jtnord@users.noreply.github.com>

* Update test/src/test/java/hudson/PluginManagerTest.java

Co-authored-by: James Nord <jtnord@users.noreply.github.com>

* Update test/src/test/java/hudson/PluginManagerTest.java

Co-authored-by: James Nord <jtnord@users.noreply.github.com>

* Update test/src/test/java/hudson/model/HelpLinkTest.java

Co-authored-by: James Nord <jtnord@users.noreply.github.com>

* Update test/src/test/java/hudson/model/HelpLinkTest.java

Co-authored-by: James Nord <jtnord@users.noreply.github.com>

* Update test/src/test/java/lib/hudson/ListScmBrowsersTest.java

Co-authored-by: James Nord <jtnord@users.noreply.github.com>

* Update test/src/test/java/lib/hudson/ListScmBrowsersTest.java

Co-authored-by: James Nord <jtnord@users.noreply.github.com>

* spotless fix

---------

Co-authored-by: James Nord <jtnord@users.noreply.github.com>

jtnord committed:
[JEP-237] introduce FIPS property (#8482)

Introduces the code that core and plugins can use to switch behaviour
based on FIPS-140 requirements

basil committed:
Remove KXML2 library (#8503)

daniel-beck committed:
Merge branch 'master' into HudsonPrivateSecurityRealmImplementation

timja committed:
Tweak changelog entry placeholder (#8532)

NotMyFault committed:
Fill in since annotations (#8530)

renovate[bot] committed:
Update dependency eslint to v8.50.0 (#8525)

renovate[bot] committed:
Update babel monorepo to v7.23.0 (#8533)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

basil committed:
[JENKINS-72067] High memory usage from `XStream2.AssociatedConverterImpl` (#8529)

dependabot[bot] committed:
Bump org.jenkins-ci.main:jenkins-test-harness from 2064.vcd3b_b_8f3f2b_a_ to 2085.va_c531db_287b_d (#8535)

Bump org.jenkins-ci.main:jenkins-test-harness

Bumps [org.jenkins-ci.main:jenkins-test-harness](https://github.com/jenkinsci/jenkins-test-harness) from 2064.vcd3b_b_8f3f2b_a_ to 2085.va_c531db_287b_d.
- [Release notes](https://github.com/jenkinsci/jenkins-test-harness/releases)
- [Changelog](https://github.com/jenkinsci/jenkins-test-harness/blob/master/docs/CHANGELOG-OLD.md)
- [Commits](https://github.com/jenkinsci/jenkins-test-harness/commits)

---
updated-dependencies:
- dependency-name: org.jenkins-ci.main:jenkins-test-harness
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

jglick committed:
Avoid saving disabled status when deleting a project (#8528)

basil committed:
Automate Java version recommendation administrative monitor (#8526)

* Automate Java version recommendation administrative monitor

* Improve dates

timja committed:
[JENKINS-70906] Remove prototype from core (#7781)

* Remove Ajax.Request from breadcrumbs

* Remove Ajax.Request from build-timeline-widget

* WIP

* Use crumb API for setting up crumb

* Use existing semantics

* Fix build button column

* Add comment explaining why catch is added

* Use post and crumb for combobox

* Lint fix

* More fixes

* Fixes

* Fix

* Fix validate button

htmlunit was a pain...

* Lint fix

* Fix URICheckEncodingMonitor

* Add a user experimental flag to run Jenkins without Prototype.js

* Adapt select.js

* Fix parameter support

* JENKINS-70922 Remove Prototype Ajax.Request usage from select.js

* Lint:fix

* More

* lint

* Rebased

* Lint

* Fix bad merge

* JENKINS-70931 Remove Prototype Ajax.Request usages from hudson-behavior.js

Co-authored-by: Basil Crow <me@basilcrow.com>

* Fix URICheckEncodingMonitor

* JENKINS-70921 Remove Prototype Ajax.Request usages from link.js

* Remove dead code

* JENKINS-70910 Remove Prototype.Browser usages from hudson-behavior.js

* JENKINS-70947 Remove Prototype $ usage from section_.js

* Prototype removal true

* Remove unneeded test

* Remove hacky test that doesn't work with fetch

* Remove unneeded workaround

* Fix test

* Minor improvements while debugging

* Fix repeatable.js

* JENKINS-71020 Remove Prototype usages from buildTimeTrend_resources.js

* lint

* JENKINS-71020 Remove Prototype usages from buildTimeTrend_resources.js

* Use escapeHTML instead of inlined references

* Fix issues

* JENKINS-70931 Remove Prototype Ajax.Request usages from hudson-behavior.js

Co-authored-by: Basil Crow <me@basilcrow.com>

* Remove hacky test that doesn't work with fetch

* Remove unused checkMethod

* Revert "Revert "JENKINS-70922 Remove Prototype Ajax.Request usage from select.js""

This reverts commit 2cff5b87b52fa80be8c9cb650724e8e3475e8a97.

* Serialise refills

Co-authored-by: Basil Crow <me@basilcrow.com>

* More reverts

* More merge

* Remove prototype.js

* Remove more references

---------

Co-authored-by: Basil Crow <me@basilcrow.com>

renovate[bot] committed:
Update dependency node to v18.18.0 (#8537)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency postcss to v8.4.31 (#8540)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency postcss-scss to v4.0.9 (#8541)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.jenkins-ci.plugins:display-url-api from 2.3.9 to 2.200.vb_9327d658781 (#8536)

Bump org.jenkins-ci.plugins:display-url-api

Bumps [org.jenkins-ci.plugins:display-url-api](https://github.com/jenkinsci/display-url-api-plugin) from 2.3.9 to 2.200.vb_9327d658781.
- [Release notes](https://github.com/jenkinsci/display-url-api-plugin/releases)
- [Changelog](https://github.com/jenkinsci/display-url-api-plugin/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jenkinsci/display-url-api-plugin/commits)

---
updated-dependencies:
- dependency-name: org.jenkins-ci.plugins:display-url-api
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump io.jenkins.plugins:plugin-util-api from 3.3.0 to 3.4.0 (#8538)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump com.puppycrawl.tools:checkstyle from 10.12.3 to 10.12.4 (#8547)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.jenkins-ci.plugins:credentials from 1271.v54b_1c2c6388a_ to 1290.v2e5b_13eb_b_127 (#8548)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

MarkEWaite committed:
Warn users of old Java version 18 months before end of life (#8543)

basil committed:
Fix encoding of Brazilian Portuguese localization (#8549)

dependabot[bot] committed:
Bump asm.version from 9.5 to 9.6 (#8545)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump org.jenkins-ci:jenkins from 1.105 to 1.106 (#8550)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

timja committed:
Tweak changelog guidance in PR template (#8553)

* Tweak changelog guidance in PR template

* Update .github/PULL_REQUEST_TEMPLATE.md

abhishekmaity committed:
add gitter link (#8539)

* missing 'alt' attribute added

* missing 'alt' attribute added

* replaced 'new URL' to 'new URI' per Java 11 target codebase

* Revert "replaced 'new URL' to 'new URI' per Java 11 target codebase"

This reverts commit aeef4c1e4bb57a4832dbe368920aa0716a3f68e7.

* add jenkins gitter chat link at README.md

jglick committed:
Allow location of `queue.xml` to be overridden (#8531)

dependabot[bot] committed:
Bump bridge-method-injector.version from 1.28 to 1.29 (#8552)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump access-modifier.version from 1.32 to 1.33 (#8551)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

jenkins-release-bot committed:
[maven-release-plugin] prepare release jenkins-2.426

jenkins-release-bot committed:
[maven-release-plugin] prepare for next development iteration

jtnord committed:
warn when password hashed with incorrect algorithm

dependabot[bot] committed:
Bump commons-io:commons-io from 2.13.0 to 2.14.0 (#8546)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

jtnord committed:
fix typos

Co-authored-by: Daniel Beck <1831569+daniel-beck@users.noreply.github.com>

jtnord committed:
Merge branch 'master' into HudsonPrivateSecurityRealmImplementation
